### PR TITLE
chore: merge multiple ivf_pq indices

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/lance/src/index/vector/ivf/io.rs [hunk 1]
```diff
@@ -75,20 +86,31 @@ pub(super) async fn write_index_partitions(
...
-            let pq_idx = part.as_any().downcast_ref::<PQIndex>().unwrap();
-            if pq_idx.code.is_some() {
-                let pq_code_arr = pq_idx.code.as_ref().unwrap().clone();
-                let pq_code_fixed_size_arr = FixedSizeListArray::try_new_from_values(
-                    pq_code_arr.as_ref().clone(),
-                    pq_idx.pq.num_sub_vectors() as i32,
-                )?;
-                pq_array.push(Arc::new(pq_code_fixed_size_arr));
-                row_id_array.push(pq_idx.row_ids.as_ref().unwrap().clone());
+        let mut pq_array: Vec<Arc<dyn Array>> = vec![];
```
Comment (chebbyChefNEQ) [inline rust/lance/src/index/vector/ivf/io.rs:89]: nit: should we reserve capacity here?

**Review by chebbyChefNEQ — APPROVED** [review]
let's add a test for this?

**Comment by eddyxu** [comment]
> let's add a test for this?
sure will do

File: rust/lance/src/index/vector/ivf/io.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance/src/index/vector/ivf/io.rs:89]: This is # of total sub index? it is less likely to be more than 100 ?

**Pull request merged**